File: packages/contract-py/src/tigr_asgi_contract/bindings.py
from ._enum_compat import StrEnum


class Binding(StrEnum):
    REST = 'rest'
    JSONRPC = 'jsonrpc'
    HTTP_STREAM = 'http.stream'
    SSE = 'sse'
    WEBSOCKET = 'websocket'
    WEBTRANSPORT = 'webtransport'

File: packages/contract-rs/src/bindings.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum Binding {
    #[serde(rename = "rest")]
    Rest,
    #[serde(rename = "jsonrpc")]
    Jsonrpc,
    #[serde(rename = "http.stream")]
    HttpStream,
    #[serde(rename = "sse")]
    Sse,
    #[serde(rename = "websocket")]
    Websocket,
    #[serde(rename = "webtransport")]
    Webtransport,
}
